### PR TITLE
Added a Dockerfile

### LINKED ISSUE #11: create dockerfile

### PR DESCRIPTION
**Pull request by MitchellBerend**
Resolves #11

This pr adds a Dockerfile which is a 2 stage build with the final image based on a distroless image with a debian 12 base. It also updates the README.md with a build command and adds a Makefile to make the build more ergonomic.

### CODE REVIEW

**Comment by MitchellBerend** [comment]
The output of the `make docker-build` command should maybe also include a `.env` volume mount since that is required at runtime. 

What do you think?

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+TAGNAME=bits-and-bytes-bot:latest
+
+.PHONY: docker-build
```
Comment (devanbenz) [inline Makefile]: Instead of using a makefile would you be opposed to using taskfile? 
https://taskfile.dev/

much easier to use IMO

**Review by devanbenz — CHANGES_REQUESTED** [review]
1 comment

**Comment by devanbenz** [comment]
> The output of the `make docker-build` command should maybe also include a `.env` volume mount since that is required at runtime.
> 
> What do you think?

How about instead of injecting the .env directly in to the docker file can you instead create a docker compose and lets handle the envs there? We're planning on deploying this to some sort of serverless hosting service so it will most likely end up handling the env vars.

File: Makefile (hunk 1 above)
Comment (MitchellBerend) [reply Makefile]: I have not used that but I can convert it.

Comment (devanbenz) [reply Makefile]: It's really easy to use - should only take a few mins to understand

**Comment by MitchellBerend** [comment]
It requires the `.env` injection at runtime as is, so a docker-compose file would still work with this PR. I kind of assumed the make command would only be ran in dev anyway.

Edit: It seems that the way the variables are read into memory expects a `.env` file regardless of if the variables already exist

**Comment by devanbenz** [comment]
> It requires the `.env` injection at runtime as is, so a docker-compose file would still work with this PR. I kind of assumed the make command would only be ran in dev anyway.
> 
> Edit: It seems that the way the variables are read into memory expects a `.env` file regardless of if the variables already exist

Ah yeah good call - I'm loading the .env file using a go library haha. It may be good to setup a "dev" switch or something that loads the .env file only when running locally.

**Comment by MitchellBerend** [comment]
A simple check for if the environment variables exist already would also work right? If any of them don't exist the program can then try and load the `.env` like it does now to read the rest maybe? It would make the .env useful for overriding variables too but I'm not sure how useful that would be.

**Comment by devanbenz** [comment]
@MitchellBerend https://github.com/devanbenz/bits-and-bytes-bot/pull/16 please check this pr and merge 
will require you to run `ENV=local go run . ` to run locally and load up .env

**Review by devanbenz — APPROVED** [review]
lgtm

**Comment by devanbenz** [comment]
will create a seperate issue to make a docker compose cause I wanna get the datastore setup so it can be added to the compose 👍

**Pull request merged**